tasks/main.yml

```
@@ -1,2 +1,84 @@
 ---
+- block:
+  - name: Determine latest GitHub release (local)
+    delegate_to: localhost
+    become: false
+    uri:
+      url: "https://api.github.com/repos/go-acme/lego/releases/latest"
+      body_format: json
+    register: _github_release
+    until: _github_release.status == 200
+    retries: 5
 
+  - name: Set certs_rfc2136_lego_version
+    set_fact:
+      certs_rfc2136_lego_version: "{{ _github_release.json.tag_name
+        | regex_replace('^v?(.*)$', '\\1') }}"
+  when: (certs_rfc2136_lego_version is not defined) or (certs_rfc2136_lego_version == 'latest')
+  run_once: true
+
+- name: Create chache directory
+  file:
+    path: /var/cache/lego/{{ certs_rfc2136_lego_version }}
+    state: directory
+
+- name: Download lego and verify SHA256 checksum
```

```suggestion
- name: Download lego
```
I'd say the hash verification is an implementation detail of the download step.